lnpbp-0031.md

```diff
@@ -40,8 +42,140 @@ License: CC0-1.0
+data TxIn             :: prevOutput Outpoint
+                       , sigScript SigScript
+                       , sequence SeqNo
+                       , witness Witness
+data TxOut            :: value Sats, scriptPubkey ScriptPubkey
+data TxVer            :: U32
+data Txid             :: [U8 ^ 32]
+data Vout             :: U32
+data Witness          :: [[U8 ^ ..0xffffffffffffffff] ^ ..0xffffffffffffffff]
+```
+
+`CommitVerify.sty`, semantic lib id `texas_year_ethnic_CPr8tcdPqWZ3KP8dXNPYavTEkbn8PG7CoJHtfwDFKRHJ`
+```haskell
+typelib CommitVerify -- version v0.10.0-2023.03.05.A
+
+data Message          :: [U8 ^ 32]
+data ProtocolId       :: [U8 ^ 32]
+data MerkleNode       :: [U8 ^ 32]
+
+data TreeNode         :: concealedNode (depth StdLib.U4, hash MerkleNode)
+                       | commitmentLeaf (protocolId ProtocolId, message Message)
+
+data MerkleTree       :: depth StdLib.U4
+                       , entropy U64
+                       , messages {[Byte ^ 32] -> Message}
+                       , map {U16 -> ProtocolId, Message}
+
+data MerkleBlock      :: depth StdLib.U4
+                       , crossSection [TreeNode]
+                       , entropy U64?
+
+data MerkleProof      :: pos U16, path [MerkleNode]
+```
+
+`RGBContracts.sty`, semantic lib id ``:
+```haskell
+typelib RGBTypes -- version v0.10.1-2023.04.20.C
+
+import panel_chamber_ohio_GWzzfwBqzA5BUVER6hqq3rBdDs6UUJ15w1T8ys6WrNr5 as Bitcoin
+import olivia_angel_micro_Fd3Lx2smvwdKhRCkhMs2du3CAYyKqJneNSf1g72WjBoq as _
+
+-- number of decimal fractions (decimal numbers after floating point)
+data Precision :: indivisible:0 
+                | deci:1 
+                | centi:2 
+                | milli:3
+                | deciMilli:4
+                | centiMilli:5 
+                | micro:6 
+                | deciMicro:7 
+                | centiMicro:8 
+                | nano:9 
+                | deciNano:10 
+                | centiNano:11 
+                | pico:12 
+                | deciPico:13 
+                | centiPico:14 
+                | femto:15 
+                | deciFemto:16 
+                | centiFemto:17 
+                | atto:18
+
+data Amount :: Zk64 -- fungible asset amount
+
+data Ticker :: [AlphaCapsNum ^ 1..8]
+data Name :: [AsciiPrintable ^ 1..40]
+data Details :: [Unicode ^ 1..256]
+
+data AssetNaming ::
+    ticker Ticker,
+    name Name,
+    details Details?
+
+data DivisibleAssetSpec ::
+    naming AssetNaming, 
+    precision Precision
+    
+data RicardianContract :: [Unicode]
+
+-- `reg-name` defined by MIME spec
+-- TODO: filter on the character level with a more precise metric
+data MediaRegName :: [AsciiPrintable ^ 1..127]
+
+data MediaType ::
+    type MediaRegName,
+    subtype MediaRegName,
+    -- We do not support other parameters
+    charset MediaRegName
+
+-- proof of reserves
+data PoR ::
+    utxo Bitcoin.Outpoint,
+    proof [Bytes] -- auxilary data which are schema-specific
```

Comment:
Shouldn't this be defined as `proof [Byte]`?

Reply:
Yes it should

```suggestion
    proof [Byte] -- auxilary data which are schema-specific
```